tests/FakeItEasy.Tests/ArgumentConstraintManagerExtensions/StringEndsWithTests.cs

```diff
@@ -45,7 +45,9 @@ public override void IsValid_should_return_true_for_valid_values(object validVal
 
         protected override void CreateConstraint(INegatableArgumentConstraintManager<string> scope)
         {
+#pragma warning disable CA1307 // Specify StringComparison
             scope.EndsWith("table");
+#pragma warning restore CA1307 // Specify StringComparison

```

Comment:
I didn't realize this warning also applied to types other than `String`. Doesn't that mean that existing user code calling this method will start showing this warning too?

Reply:
Oh, that's a very good point. I didn't think about the warning flowing through. I'll look into it.

Reply:
It seems the analyzer is looking at the shape of the methods and what sibling methods exist with or without the comparison type. 
I don't know how we could avoid triggering the warning for clients, short of either changing the signature (reorder the parameters?), or replacing the original constraint with a differently-named one ("EndsWithExactly" or similar). I don't love either option, and I don't really want to subject a bunch of presumably happy users to a new warning, especially when I think the method that causes the warning is the better choice in general.
I'm at a loss.

Reply:
>  or replacing the original constraint with a differently-named one ("EndsWithExactly" or similar)

Of renaming the new ones, I suppose.  "EndsWithUsingComparison" or similar. Unsurprisingly, I don't love this either.

Given that this was an owner-requested feature, I'm losing enthusiasm! But will still think on it.

Reply:
I don't think it's a very major issue. Only users who use FxCop will see the warning, and those users are typically used to this kind of warning and know how to fix them. It's annoying, but easy to fix.
Alternatively, we could consider this a breaking change (since it can produce new warnings in code that didn't have them), and release this change in a new major version. But it's probably overkill... I'm more inclined to just ignore the problem.